.github/workflows/build.yml

```diff
@@ -0,0 +1,60 @@
+name: build
+
+on:
+  push:
+    branches: ["main"]
+  pull_request:
+  workflow_dispatch:
+  
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    permissions: # required by aws-actions/configure-aws-credentials
+      id-token: write
+      contents: read
+
+    steps:
+    - uses: actions/checkout@v3
+
+    - uses: actions/setup-java@v3
+      with:
+        distribution: 'corretto'
+        java-version: '11'
+        cache: 'sbt'
+
+    - uses: actions/setup-node@v3
+      with:
+        node-version-file: .nvmrc
+
+    - name: CI
+      working-directory: cdk
+      run: |
+        yarn install --frozen-lockfile
+        yarn lint
+        yarn test
+        yarn synth
+
+    - name: AWS Auth
+      uses: aws-actions/configure-aws-credentials@v2
+      with:
+        role-to-assume: ${{ secrets.GU_RIFF_RAFF_ROLE_ARN }}
+        aws-region: eu-west-1
+
+    - name: Build and Test project, Assemble jar, copy to root dir
+      run: |
+        sbt clean compile test assembly
+        cp target/scala*/elastic-search-monitor.jar ./elastic-search-monitor.jar
+
+    - name: Upload to riff-raff
+      uses: guardian/actions-riff-raff@v2
+      with:
+        app: elastic-search-monitor
```

Comment:
minor: From [the docs](https://github.com/guardian/actions-riff-raff#available-inputs), `app` isn't needed when `projectName` is also set. In this case, I think `projectName` is the correct variable to use, as the current project doesn't use the `stack::app` format.